packages/host/app/components/search-sheet/search-result/index.gts

```diff
@@ -0,0 +1,41 @@
+import Component from '@glimmer/component';
+import type { Card } from 'https://cardstack.com/base/card-api';
+import { cardTypeDisplayName } from '@cardstack/runtime-common';
+import { CardContainer } from '@cardstack/boxel-ui';
+import { trackedFunction } from 'ember-resources/util/function';
+import type CardService from '../../../services/card-service';
+import { service } from '@ember/service';
+
+interface Signature {
+  Args: {
+    card: Card;
+  };
+}
+
+export default class SearchResult extends Component<Signature> {
+  <template>
+    <CardContainer @displayBoundaries={{true}} class='search-result' data-test-search-result={{@card.id}}>
+      <header class='search-result__title'>{{@card.title}}</header>
+      <p class='search-result__display-name'>{{cardTypeDisplayName @card}}</p>
+      <p class='search-result__realm-name'>In {{this.realmName}}</p>
+    </CardContainer>
+    <style>
+      .search-result { padding: var(--boxel-sp); width: 250px; }
+      .search-result__title { margin-bottom: var(--boxel-sp-xs); font: 500 var(--boxel-font-sm); overflow: hidden; text-wrap: nowrap; }
+      .search-result__display-name { margin: 0; font: 500 var(--boxel-font-xs); color: #919191}
+      .search-result__realm-name { margin: 0; color: var(--boxel-teal); font-size: var(--boxel-font-size-xs); }
+    </style>
+  </template>
+
+  @service declare cardService: CardService;
+
+  fetchRealmName = trackedFunction(this, async () => {
+    let realmInfoSymbol = await this.cardService.realmInfoSymbol();
```

Comment:
its worth noting that this async in the component is safe because the `trackedFunction()` takes in the parent component (`this`) and entangles the lifetime of the tracked function with the parent component so that if the parent component is destroyed the tracked function is as well.